packages/docusaurus-mdx-loader/src/remark/toc/search.js

```diff
@@ -44,28 +40,57 @@ function search(node) {
   const onHeading = (child, index, parent) => {
     const value = toString(child);
 
-    if (parent !== node || !value || child.depth > 3 || child.depth < 2) {
+    if (
+      parent !== headingNode ||
```

Comment:
Is it possible (either as part of this PR or a future one) to add a config option to remove this part of the conditional?

I have a use case where I want headings that aren't at the root of the document (in this case, inside a list element) to show in the TOC.

Reply:
Can you explain better your use-case with an example source doc and an example TOC we should compute? (ie provide a test case somehow?)

If we use min/max heading levels, does it work for your use-case? (like min=3, max=4 for example?)

CF https://github.com/facebook/docusaurus/pull/5578

Reply:
Sure, see https://raw.githubusercontent.com/australiangreens/agvrules/master/docs/proposed-constitution/01-fundamental-matters.md

The TOC I would like is:

* Purposes
* Policies and party strategy
* Public office
* Decision-making
* Openness
* Affirmative action

Without the suggested change (which I am implementing in the form of [a patch](https://github.com/australiangreens/agvrules/blob/master/patches/%40docusaurus%2Bmdx-loader%2B2.0.0-beta.6.patch)), I get no TOC at all, because all the headings are not top-level elements.

Reply:
I'm not really able to run your doc, can I see a live published version of it too?

What I understand is that you still want min=2 for the toc headings, but you need headings that are part of a list?

> 1. ## Purposes

This is not really an use-case we had in mind afaik. Why can't your readings be top-level and are part of the list?

Is this the intended behavior?

![image](https://user-images.githubusercontent.com/749374/134527101-864141d7-5e4d-43c9-84d6-6e750d0ebc29.png)

Can you explain your motivations for this unusual headings usage?

Reply:
The published version is at https://australiangreens.github.io/agvrules/proposed-constitution/fundamental-matters/

The headings are part of the list item because:

* that's how they appear in the source document I'm copying 
* that's semantically correct 
* that structure means that things that parse lists read it correctly

But I think there's also much simpler situations where a heading might not appear at the root of a document. For example, if it's inside a quote block or an admonition. And I can see cases where people would want those to show in the TOC.

Reply:
Thanks @ineffyble

As the TOC PR is already complex enough, I'd like to support your use-case in another one

Can you please open another dedicated issue with all those relevant info?

> For example, if it's inside a quote block or an admonition.

I'd find this weird to use a heading this way 😅 

> And I can see cases where people would want those to show in the TOC.

I'd be very interested to have a list of use-cases that you have in mind where the current logic is not good enough.

Passing a function as TOC config is the most flexible, but it may not be an ideal API to deal with and would be hard to document, so I'm not sure this is what we want. If we supported your use-cases by default it could be good enough?